### PR TITLE
fix: Decode spaces in factory url

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Decode spaces in factory url

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/22458

### Is it tested? How?

N/A

#### Release Notes

N/A

### CODE REVIEW

**Review by olexii4 — APPROVED** [review]
LGTM

File: packages/dashboard-backend/src/routes/factoryAcceptanceRedirect.ts [hunk 1]
```diff
@@ -37,6 +37,10 @@ export function registerFactoryAcceptanceRedirect(instance: FastifyInstance): vo
           params.append(ERROR_CODE_ATTR, querystring.unescape(query[ERROR_CODE_ATTR] as string));
         }
 
+        const url = params.get('url');
+        if (url) {
+          params.set('url', url.replaceAll(' ', '%20'));
```
Comment (ibuziuk) [inline packages/dashboard-backend/src/routes/factoryAcceptanceRedirect.ts:42]: can we use `encodeURIComponent` to cover the general case? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

**Review by akurinnoy — APPROVED** [review]
LGTM

**Review by ibuziuk — APPROVED** [review]
I'm ok to merge as a hot fix for 3.10, but not really understand why do we decoding twice in general🤷

**Review by akurinnoy — APPROVED** [review]

**Pull request merged**